### PR TITLE
Update BuildKit for cache key memory usage patch

### PR DESCRIPTION
**Pull request by mikejholly**
Updates BK for memory usage fixes in https://github.com/earthly/buildkit/pull/33

### CODE REVIEW

**Comment by mikejholly** [comment]
Note: I ran the repro case for [ issue #2957](https://github.com/earthly/earthly/issues/2957) and did not encounter the error.

```
date > my-file; earthly-dev +foo & earthly-dev +foo
```

File: examples/bazel/Earthfile [hunk 1]
```diff
@@ -27,8 +27,8 @@ build:
     FROM +code
     # Save Bazel's cache as a cache directory. This directory will be reused
     # between runs.
-    ARG bazel_cache = "/root/.cache/bazel"
-    CACHE "$bazel_cache"
+    # ARG bazel_cache = "/root/.cache/bazel"
+    # CACHE "$bazel_cache"
```
[the remark this answers is not in the document]
Comment (mikejholly) [reply examples/bazel/Earthfile:31]: Yes, but I'm not sure why these lines were leading to this failure: https://github.com/earthly/earthly/actions/runs/7023739400/job/19112169282

```
./e/bazel+build *failed* | --> expandargs readlink -f ./bazel-out
./e/bazel+build *failed* | [no output]
./e/bazel+build *failed* | ERROR examples/bazel/Earthfile line 33:4
./e/bazel+build *failed* |       The command
./e/bazel+build *failed* |           expandargs readlink -f ./bazel-out
./e/bazel+build *failed* |       did not complete successfully. Exit code 1
```

[the remark this answers is not in the document]
Comment (vladaionescu) [reply examples/bazel/Earthfile:31]: Perhaps we should just remove this example altogether. It's not very polished anyway.

[the remark this answers is not in the document]
Comment (mikejholly) [reply examples/bazel/Earthfile:31]: Either or. The target seems to work without that `CACHE` command.

[the remark this answers is not in the document]
Comment (vladaionescu) [reply examples/bazel/Earthfile:31]: Yeah.. well the `CACHE` was the most important point of the example.

[the remark this answers is not in the document]
Comment (mikejholly) [reply examples/bazel/Earthfile:31]: Ok, I can look into it further.

**Comment by alexcb** [comment]
I gave this some some more stress testing; by running:

```
VERSION 0.7

test1:
  FROM alpine:3.17
  COPY my-file .
  RUN true
  RUN md5sum my-file > data
  SAVE ARTIFACT data data

test2:
  FROM alpine:3.17
  COPY my-file .
  RUN true

test3:
  FROM alpine:3.17
  COPY +test1/data .
  RUN cat data

test4:
  FROM +test3
  IF ls data
    BUILD +test1
  END

all:
  BUILD +test1
  BUILD +test2
  BUILD +test3
  BUILD +test4
```

along with

```
for i in $(seq 1 100); do /home/alex/gh/earthly/earthly/build/linux/amd64/earthly +all & done
```

and another process periodically changing the contents of `my-file`:

```
while sleep 0.1; do uuidgen > my-file; done
```

and didn't see any solver issues, so I'm feeling less worried about this change :)

**Comment by mikejholly** [comment]
@alexcb Cool. Thanks for doing that. I've also been running our Core tests against this branch with no apparent issues. https://github.com/earthly/earthly/actions/runs/7024593005?pr=3505

**Pull request closed without merging**